docs/root/configuration/listeners/network_filters/zookeeper_proxy_filter.rst

```diff
@@ -70,6 +97,33 @@ The following counters are available:
   getallchildrennumber_rq, Counter, Number of getallchildrennumber requests
   close_rq, Counter, Number of close requests
   response_bytes, Counter, Number of bytes in decoded response messages
+  connect_resp_bytes, Counter, Number of bytes in decoded connect response messages
+  ping_resp_bytes, Counter, Number of bytes in decoded ping response messages
+  auth_resp_bytes, Counter, Number of bytes in decoded auth response messages
+  getdata_resp_bytes, Counter, Number of bytes in decoded getdata response messages
+  create_resp_bytes, Counter, Number of bytes in decoded create response messages
+  create2_resp_bytes, Counter, Number of bytes in decoded create2 response messages
+  createcontainer_resp_bytes, Counter, Number of bytes in decoded createcontainer response messages
+  createttl_resp_bytes, Counter, Number of bytes in decoded createttl response messages
+  setdata_resp_bytes, Counter, Number of bytes in decoded setdata response messages
+  getchildren_resp_bytes, Counter, Number of bytes in decoded getchildren response messages
+  getchildren2_resp_bytes, Counter, Number of bytes in decoded getchildren2 response messages
+  delete_resp_bytes, Counter, Number of bytes in decoded delete response messages
+  exists_resp_bytes, Counter, Number of bytes in decoded exists response messages
+  getacl_resp_bytes, Counter, Number of bytes in decoded getacl response messages
+  setacl_resp_bytes, Counter, Number of bytes in decoded setacl response messages
+  sync_resp_bytes, Counter, Number of bytes in decoded sync response messages
+  check_resp_bytes, Counter, Number of bytes in decoded check response messages
+  multi_resp_bytes, Counter, Number of bytes in decoded multi response messages
+  reconfig_resp_bytes, Counter, Number of bytes in decoded reconfig response messages
+  setwatches_resp_bytes, Counter, Number of bytes in decoded setwatches response messages
+  setwatches2_resp_bytes, Counter, Number of bytes in decoded setwatches2 response messages
+  addwatch_resp_bytes, Counter, Number of bytes in decoded addwatch response messages
+  checkwatches_resp_bytes, Counter, Number of bytes in decoded checkwatches response messages
+  removewatches_resp_bytes, Counter, Number of bytes in decoded removewatches response messages
+  getephemerals_resp_bytes, Counter, Number of bytes in decoded getephemerals response messages
+  getallchildrennumber_resp_bytes, Counter, Number of bytes in decoded getallchildrennumber response messages
+  close_resp_bytes, Counter, Number of bytes in decoded close response messages
```

Comment:
High-level comment - stats are not free, and it seems that many are added here, which may impact performance. Have you considered other mechanisms to achieve your goal?

Reply:
Although ZooKeeper 3.6.0 introduced [a new metric system](https://zookeeper.apache.org/doc/current/zookeeperMonitor.html#:~:text=Four%20letter%20words-,New%20Metrics%20System,%2C%20requestProcessor%2C%20and%20so%20forth.), it does not provide the per opcode request/response bytes metrics, which could be helpful to understand the payload size of different opcodes during debugging. This PR also introduces switch(s) to enable/disable these metrics. Then users could stop emitting these metrics if not needed.

Reply:
Yeah, you are right! Let me refactor the code to initial new metrics in filter config.

Reply:
FWIW, I don't think the refactor addressed the issue of Envoy counters being initialized for every instantiation of the filter.

Reply:
My understanding is that these new counters are added to the filter config, which will be shared by all filters.
Not sure whether you have some examples that I could refer to, thanks